### PR TITLE
Refactor/update-contributing-guide

### PR DESCRIPTION
**Pull request by alifeee**
Update contributing guide

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,16 +176,16 @@ worksheet.batch_update([{
 
 ## [Documentation](https://gspread.readthedocs.io/en/latest/)
```
Comment (lavigne958) [inline README.md:177]: I think as we are currently updating that part we can change something too here:

- I always get confused when I see a link on the header of a markdown file.
- We could leave the header with `Documentation`
- and move the actual link to a basic paragraph bellow like "Documentation can be found here" with the link on "here"

What do you think ?

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,121 +1,120 @@
...
-```
+### `GS_CREDS_FILENAME` environment variable
 
-Then run the tests in offline mode to make sure you have recorded everything.
+You must provide service account credentials using the `GS_CREDS_FILENAME` environment variable in order to make HTTP requests to the Sheets API.
 
-```
-tox -e py
-```
+[Obtain OAuth2 credentials from Google Developers Console](http://gspread.readthedocs.org/en/latest/oauth2.html).
```
Comment (lavigne958) [inline .github/CONTRIBUTING.md:81]: In the record mode we expect a service account in order to run the test onlie.

The correct link is: `https://docs.gspread.org/en/latest/oauth2.html#for-bots-using-service-account`

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,121 +1,120 @@
...
 
-## Render Documentation
+### Run test, capturing *only new* HTTP requests
 
-The documentation uses [reStructuredText](http://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html#rst-index) markup and is rendered by [Sphinx](http://www.sphinx-doc.org/).
+To record new HTTP requests:
 
-To build the documentation locally, use the following command:
+1. Remove the file holding the recorded HTTP requests of the test(s).
+  (e.g.: for the file `tests/cell_test.py` delete `tests/cassettes/CellTest.json`)
```
Comment (lavigne958) [inline .github/CONTRIBUTING.md:110]: we create a single cassette file for a single test, so in the `cell_test.py` file we have multiple tests, so it produces multiple cassettes.

Here in order to guide the new user perfectly we should be more prices and mention:
- for the file `tests/cell_test.py`
- for the test `test_a1_value`
- remove the file: `tests/cassettes/CellTest.test_a1_value.json`

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,121 +1,120 @@
...
 
-**Note:** the CI runs that command, if it fail you won't be able to merge
-your changes in GSpread.
+Installing all the `*.txt` files is optional (tox installs dependencies per-run).
 
-```
-tox -e py
+```bash
+pip install tox
+pip install -r ./test-requirements.txt -r ./lint-requirements.txt -r ./docs/requirements.txt
```
Comment (lavigne958) [inline .github/CONTRIBUTING.md:32]: Interestingly enough we don't require to install the dependency files. Tox will install them in a virtual env :smiley: 

The user is only *required* to installed `tox` then the rest is just useful for your editor to find the libraries installed on your computer for development purpose (mostly google dependencies)

**Review by lavigne958 — COMMENTED** [review]
Thank you for this awesome contribution ! 

This is much easier to understand, I only have a few comments then we after fixing them we can merge it !

**Comment by alifeee** [comment]
ready to merge! click tha button! 🚀

**Pull request merged**